src/ansys/dpf/core/mesh_info.py

```diff
@@ -0,0 +1,193 @@
+"""
+MeshInfo
+==========
+"""
+import ansys.dpf.core as dpf
+from ansys.dpf.core import server as server_module
+from ansys.dpf.core.generic_data_container import GenericDataContainer
+
+
+class MeshInfo:
+    """Represents the mesh information.
+
+    This class describes the available mesh information.
+
+    Parameters
+    ----------
+     server : ansys.dpf.core.server, optional
+        Server with the channel connected to the remote or local instance.
+        The default is ``None``, in which case an attempt is made to use the
+        global server.
+
+    Examples
+    --------
+    Explore the mesh info from the model
+
+    >>> from ansys.dpf import core as dpf
+    >>> from ansys.dpf.core import examples
+    >>> fluent = examples.fluid_axial_model()
+    >>> model = dpf.Model(fluent)
+    >>> mesh_info = model.metadata.mesh_info_provider # printable mesh_info
+
+    """
+
+    def __init__(
+        self,
+        generic_data_container: GenericDataContainer = None,
+        mesh_info: dpf.MeshInfo = None,
```

Comment:
`mesh_info: MeshInfo = None` should work (it's in the same file, so it is in the namespace)

Reply:
No it doesn't with a "Unresolved reference 'MeshInfo'"

Reply:
@MichaelNale I can explain how to typehint the class you are writing

Reply:
At the end I just removed it. I should put this in place to have best Python practice I guess?

Reply:
To be able to typehinting something that is being defined, add:
`from __future__ import annotations`
in your imports.
You should then be able to typehint `MeshInfo` while in `MeshInfo`